pyproject.toml

```diff
@@ -0,0 +1,35 @@
+[tool.poetry]
+name = "bqrun"
+version = "0.0.0"  # use poetry-dynamic-versioning
+description = "Query runner for BigQuery. It automatically analyzes dependencies and runs only necessary queries in parallel."
+authors = ["Yasunori Horikoshi <horikoshi.et.al@gmail.com>"]
+readme = "readme.rst"
+license = "MIT"
+repository = "https://github.com/hotoku/bqrun"
+keywords = ["bigquery"]
+
+
+[tool.poetry-dynamic-versioning]
+enable = true
+style = "pep440"
```

Comment:
This plugin enables us to dynamic versioning based on tags on git repository.

Reply:
Thank you for your proposal of a new tool. I have not heard about it but understand it's useful. I agree starting to use the plugin.

But before moving to poetry-dynamic-versioning, I think there needs several modifications, yet.

1. You have inserted the version placeholder for it in `pyproject.toml`. But there are another version number [here](https://github.com/hotoku/bqrun/blob/main/bqrun/__init__.py#L1).
2. The plugin's official repository's sample configuration of `build-system.requires` of `pyproject.toml` loooks `["poetry-core>=1.0.0", "poetry-dynamic-versioning"]` but [your request](https://github.com/hirosassa/bqrun/blob/0d5211a877bca8be6310efb9feb8c25183ace484/pyproject.toml#L34) contains only `["poetry-core>=1.0.0"]`. Is this intended? Sorry, I'm new to the plugin, so I cannot understand..

Reply:
@hirosassa Do you have time for the above modifications? If it's difficult, I'm OK to merge your PR as is, I will apply the modification after the merge commit.

Reply:
Thanks for your comment.
This is from [README](https://github.com/mtkennerly/poetry-dynamic-versioning/blob/master/README.md) of the official repo.

> Poetry's typical version setting is still required in [tool.poetry], but you are encouraged to use version = "0.0.0" as a standard placeholder.

Since [tool.poetry section in pyproject.toml requires `version` settings, we have to set placeholder.

> The plugin's official repository's sample configuration of build-system.requires of pyproject.toml loooks ["poetry-core>=1.0.0", "poetry-dynamic-versioning"]

Yeah, your suggestion is right. I didn't know that.
This change is added in [this commit](https://github.com/mtkennerly/poetry-dynamic-versioning/commit/c8e6ac0ced0ff5140166321cae1f7f0f9922fb12).

For example of [gokart repo](https://github.com/m3dev/gokart/blob/master/pyproject.toml) (I'm a maintainer of the repo), we didn't write `["poetry-core>=1.0.0", "poetry-dynamic-versioning"]`.
But this change is not tested, so I'll take a look.

Reply:
About 1. of my comment.

Sorry, my explanation was ambiguous and my point was missed.

In this package, there are 2 places where the version number is referred.

1. in `/pyproject.toml`: `[tool.poetry] version = ...` [here](https://github.com/hirosassa/bqrun/blob/0d5211a877bca8be6310efb9feb8c25183ace484/pyproject.toml#L3)
2. in `/bqrun/__init__.py`: `__version__ = ...` [here](https://github.com/hirosassa/bqrun/blob/0d5211a877bca8be6310efb9feb8c25183ace484/bqrun/__init__.py#L1)

In your PR, the former is written as placeholder ("0.0.0"), but the latter is left as the concrete number ("1.3.1").

To my understand, the latter should be written as placeholder

Reply:
Oh, it's my misunderstanding. 
> in `/bqrun/__init__.py`: `__version__ = ...` 

This should be removed.

Reply:
@hotoku I tested my local env and found it that `["poetry-core>=1.0.0", "poetry-dynamic-versioning"]` this entry is not necessary, but it is described on the README on official repo, so I added in 6413a73.

Also, I removed version setting on `__init__.py` in 99f6b53